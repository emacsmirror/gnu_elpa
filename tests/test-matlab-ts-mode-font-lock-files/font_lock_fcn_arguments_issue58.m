% Take from myBar in
% https://www.mathworks.com/help/matlab/matlab_prog/validate-name-value-arguments.html
%
% See: https://github.com/acristoffers/tree-sitter-matlab/issues/58

function font_lock_fcn_arguments_issue58(x,y,propArgs)
    arguments
        x (:,:) double
        y (:,:) double
        propArgs.?matlab.graphics.chart.primitive.Bar
        propArgs.FaceColor {mustBeMember(propArgs.FaceColor,{'red','blue'})} = "blue"
    end
    propertyCell = namedargs2cell(propArgs);
    bar(x,y,propertyCell{:})
end
