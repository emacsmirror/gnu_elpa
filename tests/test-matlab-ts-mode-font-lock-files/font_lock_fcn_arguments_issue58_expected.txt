c ccc ccccccccc ccc

c cccc cccc ccccc cc
c cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
c
c cccc cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc

kkkkkkkk fffffffffffffffffffffffffffffffbvDvDvvvvvvvvb
    kkkkkkkkk
        P bDDDb tttttt
        P bDDDb tttttt
        ddddddddooBBBBBBDddddddddDdddddDdddddddddDddd
        PPPPPPPPDPPPPPPPPP bBBBBBBBBBBBBbddddddddDdddddddddDbSsssSDSssssSbbb o SssssS
    kkk
    vvvvvvvvvvvv o BBBBBBBBBBBBBBbddddddddbD
    BBBbdDdDddddddddddddbDbb
kkk
